docs/reference/docs/data-replication.asciidoc

```diff
@@ -0,0 +1,136 @@
+master knows that the primary is the only single good copy. We are therefore guaranteed that the master will not promote
+any other (out-of-date) shard copy to be a  new primary and that any operation indexed into the primary will not be lost.
+Of course, since at that point we run with only single copy of the data, physical hardware issues can cause data loss
+(see <<index-wait-for-active-shards>> for some mitigation options).
+
+=== Basic read model
+
+Reads in Elasticsearch can be very lightweight lookups by ID or a heavy search request with complex aggregations that
+take non-trivial CPU power. One of the beauties of the primary-backup model is that it keeps all shard copies identical
+(with the exception of in-flight operations).. As such, a single copy is sufficient to serve a read requests.
+
+When a read request is sent to a node, that node is responsible for forwarding it to the right places, collating the
+responses, and responding to the client. We call that node the coordinating node for that search. That node does so
+according to this basic flow:
+
+. Resolve the read requests to the relevant shards. Note that since most searches will be sent to one index or more,
+   they typically need to read from multiple shards, each representing a different subset of the data.
+. For each relevant shard, select an active copy from the shard replication group. This can be either the primary or
+   a replica. By default, Elasticsearch will simply round robin between the shard copies.
+. Send shard level read requests to the selected copies.
+. Combine the results and respond. Note that in the case of get by ID look up, only one shard is relevant and this step can be skipped.
+
+==== Failure handling
+
+When a shard fails to respond to a read request, the coordinating node will select another copy from the same replication group
+and send the shard level search request to that copy instead. Repetitive failures can result in no shard copies being available.
+In such cases Elasticsearch will prefer to respond fast, albeit with partial results, instead of waiting for the issue to be
+resolved (partial results are indicated in the `_shards` header of the response).
+
+=== A few simple implications
+
+Each of these basic flows determines how Elasticsearch behaves as a system for both reads and writes. Furthermore, since read
+and write requests can be executed concurrently, these two basic flows interact with each other. This has a few inherent implications:
+
+Efficient reads:: under normal operation each read operation is performed once for each relevant replication group.
+   Only under failure conditions do multiple copies of the same shard execute the same search.
+
+Read uncommitted:: since the primary first indexes locally and then replicates the request, it is possible for a
+  concurrent read to already see the change before it has been acknowledged.
```

Comment:
I think "read uncommitted" is the official term.  alternatively "read unacknowledged"?